### DISCUSSION TITLE
Recommended workaround for deprecated Column.copy()?

### DISCUSSION THREAD

**Discussion by gunziptarball** [sqlalchemy/sqlalchemy#8213] [Usage Questions] [answered]
Hello!  While cleaning up some warnings in a project of mine, I stumbled across the deprecation warning since v1.4 that `Column.copy()` will be removed in a future release.  @zzzeek states [in this comment](https://github.com/sqlalchemy/sqlalchemy/issues/5953#issue-810301180)  that using `Column.copy()` is improper.

I found another user experiencing the same issue ( https://github.com/LANL-Seismoacoustics/pisces/issues/44) when trying to accomplish the same thing I am, which is basically "cloning" tables as well

So, my question: what is the recommended way to "clone" a column object?  I realize I could just call `_copy()` instead, but a leading underscore denotes a "protected" member.

**Comment by CaselIT** [comment, marked as answer]
Hi,

If your objective it to clone a complete table maybe you can use the `table.to_metadata` method?

**Reply by gunziptarball** [reply]
That is really useful to know, thank you @CaselIT!

**Reply by jeromerg** [reply]
Could you explain how?

**Reply by CaselIT** [reply]
the docs provides examples https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.Table.to_metadata
